stm32_base/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -245,107 +238,139 @@ static void init_timer_2(size_t prescaler, unsigned int slowTimerMode)
+        }
 }
 
-int32_t timer_device_init(size_t channel, uint32_t speed,
-                          uint32_t slowTimerMode)
+void reset_device_state(const size_t chan)
 {
+        if (chan >= MK2_TIMER_CHANNELS)
+                return;
 
-    size_t prescaler = speed_to_prescaler(speed);
-    switch (channel) {
-    case 0:
-        init_timer_0(prescaler, slowTimerMode);
-        return 1;
-    case 1:
-        init_timer_1(prescaler, slowTimerMode);
-        return 1;
-    case 2:
-        init_timer_2(prescaler, slowTimerMode);
-        return 1;
-    default:
-        return 0;
-    }
+        struct state *s = &g_state[chan];
+
+        s->period = 0;
+        s->duty_cycle = 0;
+        s->q_period_ticks = 0;
 }
 
-void timer_device_reset_count(size_t channel)
+bool timer_device_init(const size_t channel, const uint32_t speed,
+                       const uint16_t quiet_period_us)
 {
-    if (channel >= 0 && channel < TIMER_CHANNELS) {
-        g_timer_counts[channel] = 0;
-    }
+        if (channel >= MK2_TIMER_CHANNELS)
+                return false;
+
+        struct config *c = &g_config[channel];
+        c->prescaler = speed_to_prescaler(speed);
+        c->q_period_us = quiet_period_us;
+
+        reset_device_state(channel);
+
+        /*
+         * TIM9 (logical timer 1) has a clock rate 2x that of the other
+         * timers.  Thus we compensate for that by doubling the prescalar
+         * value.
+         */
+        if (1 == channel)
+                c->prescaler <<= 1;
+
+
+
+        switch (channel) {
+        case 0:
+                init_timer_0(c);
+                return true;
+        case 1:
+                init_timer_1(c);
+                return true;
+        case 2:
+                init_timer_2(c);
+                return true;
+        }
+
+        /* Should never get here */
+        return false;
 }
 
+void timer_device_reset_count(size_t channel) {}
+
 uint32_t timer_device_get_count(size_t channel)
 {
-    if (channel >= 0 && channel < TIMER_CHANNELS) {
-        return g_timer_counts[channel];
-    } else {
         return 0;
-    }
 }
 
+uint32_t ticks_to_us(uint16_t ticks)
+{
+        /* 1 tick == 2us */
+        return (uint32_t) ticks << 1;
+}
 
 uint32_t timer_device_get_usec(size_t channel)
 {
-    return timer_device_get_period(channel) * 10000 / 5000;
+        return ticks_to_us(timer_device_get_period(channel));
 }
 
 uint32_t timer_device_get_period(size_t channel)
 {
-    size_t period = 0;
-    switch (channel) {
-    case 0:
-        period = timer0_period;
-        break;
-    case 1:
-        period = timer1_period;
-        break;
-    case 2:
-        period = timer2_period;
-        break;
-    default:
-        break;
-    }
-    return period;
+        return channel < MK2_TIMER_CHANNELS ? g_state[channel].period : 0;
+}
+
+
+/*
+ * = = = IRQ methods below this point = = =
+ */
+
+void update_device_state(const size_t chan, const uint16_t p_ticks,
```

Comment:
please make the p_ticks and p_h_ticks variables more verbose, or document the function. That would help